test/powershell/Modules/Microsoft.PowerShell.Utility/NewTemporaryFile.Tests.ps1

```diff
@@ -14,17 +14,16 @@
         A FileInfo object for the temporary file is returned.
 #>
 
-Describe "NewTemporaryFile" -Tags "CI" {
+Describe "New-TemporaryFile" -Tags "CI" {
 
     It "creates a new temporary file" {
-        $tempFile = New-TemporaryFile
+        try {
+            $tempFile = New-TemporaryFile
 
-        Test-Path $tempFile | Should be $true
-        $tempFile | Should BeOfType System.IO.FileInfo
-
-        if(Test-Path $tempFile)
-        {
-            Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
+            $tempFile | Should Exist
+            $tempFile | Should BeOfType System.IO.FileInfo
+        } finally {
+            [System.IO.File]::Delete($tempFile)
```

Comment:
Previous `Test-Path` and `Remove-Item` was better.

Reply:
Why? The .net method is only one line because it internally performs the check if the file exists and does not throw in this case.

Reply:
The method can throw https://msdn.microsoft.com/en-us/library/system.io.file.delete(v=vs.110).aspx?f=255&mspperror=-2147217396#Anchor_1